lib/querystring.js

```diff
@@ -236,6 +240,11 @@ function stringify(obj, sep, eq, options) {
     (encode === qsEscape ? encodeStringified : encodeStringifiedCustom);
 
   if (obj !== null && typeof obj === 'object') {
+    // eslint-disable-next-line node-core/prefer-primordials
+    if (FunctionPrototypeSymbolHasInstance(Map, obj)) {
```

Comment:
Using `FunctionPrototypeSymbolHasInstance` would fail to recognize a `Map` instance coming from a different realm. Instead, you should be using `internalBinding('types').isMap`.